lib/proto/proto.go

```diff
@@ -409,6 +408,45 @@ func setField(msg protoreflect.Message, fdesc protoreflect.FieldDescriptor, valu
 		return nil
 	}
 
+	if fdesc.IsMap() {
+		mapping, ok := value.(starlark.IterableMapping)
+		if !ok {
+			return fmt.Errorf("in map field %s: expected mappable type, but got %s", value.Type())
+		}
+
+		iter := mapping.Iterate()
+		defer iter.Done()
+
+		// Each value is converted using toProto as usual, passing the key/value
+		// field descriptors to check their types.
+		mutMap := msg.Mutable(fdesc).Map()
+		var k starlark.Value
+		for iter.Next(&k) {
```

Comment:
This might be cleaner if you  called mapping.Items() (though of course that does allocate 2*len values temporarilly).

Reply:
I decided to go for the iterator route, as the list code is also using iterators and I wanted to avoid the extra allocations (Rust-brain shining through :)).

The difference is not that large, and it doesn't reduce the verbosity inside the loop, so I would prefer to leave it as is. If you feel strongly the other way, I'm happy to change it.